libqtile/core/manager.py

```diff
@@ -1138,14 +1139,24 @@ def cmd_spawn(self, cmd, shell=False):
 
             spawn(["xterm", "-T", "Temporary terminal"])
         """
+        to_lookup = ""
         if shell:
             if not isinstance(cmd, str):
+                to_lookup = cmd[0]
                 cmd = subprocess.list2cmdline(cmd)
+            else:
+                to_lookup = cmd
             args = ["/bin/sh", "-c", cmd]
         elif isinstance(cmd, str):
             args = shlex.split(cmd)
+            to_lookup = args[0]
         else:
             args = list(cmd)
+            to_lookup = args[0]
```

Comment:
maybe worth refactoring to look for the type of `cmd` before checking `shell`

Reply:
Yeah, that might be a bit better, I'll take a look.

Reply:
So, I couldn't figure out how to do this nicely. Turns out that if `shell=True`, we need to render `cmd` as a string, but if `shell=False`, we want it as a list. I'm open to other implementations, though.

Reply:
This is the best I could come with:

```python
if isinstance(cmd, str):
    args = shlex.split(cmd)
    cmd = subprocess.list2cmdline(args)
else:
    args = list(cmd)

to_lookup = args[0]
if shell:
    args = ["/bin/sh", "-c", cmd]
```

Reply:
Cool, thanks, I updated it to this.